src/server/v2.0/handler/scanexport.go

```diff
@@ -0,0 +1,202 @@
+	BaseAPI
+	scanDataExportCtl scandataexport.Controller
+	sysArtifactMgr    systemartifact.Manager
+}
+
+func (se *scanDataExportAPI) Prepare(ctx context.Context, operation string, params interface{}) middleware.Responder {
+	return nil
+}
+
+func (se *scanDataExportAPI) ExportScanData(ctx context.Context, params operation.ExportScanDataParams) middleware.Responder {
+	if err := se.RequireAuthenticated(ctx); err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	// check if the MIME type for the export is the Generic vulnerability data
+	if params.XScanDataType != v1.MimeTypeGenericVulnerabilityReport {
+		error := &models.Error{Message: fmt.Sprintf("Unsupported MIME type : %s", params.XScanDataType)}
+		errors := &models.Errors{Errors: []*models.Error{error}}
+		return operation.NewExportScanDataBadRequest().WithPayload(errors)
+	}
+
+	// loop through the list of projects and validate that scan privilege and create privilege
+	// is available for all projects
+	// TODO : Should we just ignore projects that do not have the required level of access?
+
+	projects := params.Criteria.Projects
+	for _, project := range projects {
+		if err := se.RequireProjectAccess(ctx, project, rbac.ActionCreate, rbac.ResourceScan); err != nil {
+			return se.SendError(ctx, err)
+		}
+	}
+
+	scanDataExportJob := new(models.ScanDataExportJob)
+
+	// vendor id associated with the execution can be set to -1
+	// is not associated with the state of any other scan report entity
+	userContext := context.WithValue(ctx, "vendorId", -1)
+	secContext, err := se.GetSecurityContext(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	jobId, err := se.scanDataExportCtl.Start(userContext, se.convertToCriteria(params.Criteria, secContext.GetUsername()))
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	scanDataExportJob.ID = jobId
+	return operation.NewExportScanDataOK().WithPayload(scanDataExportJob)
+}
+
+func (se *scanDataExportAPI) GetScanDataExportExecution(ctx context.Context, params operation.GetScanDataExportExecutionParams) middleware.Responder {
+	err := se.RequireAuthenticated(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	execution, err := se.scanDataExportCtl.GetExecution(ctx, params.ExecutionID)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	sdeExec := models.ScanDataExportExecution{
+		EndTime:     strfmt.DateTime(execution.EndTime),
+		ID:          execution.ID,
+		StartTime:   strfmt.DateTime(execution.StartTime),
+		Status:      execution.Status,
+		StatusText:  execution.StatusMessage,
+		Trigger:     execution.Trigger,
+		UserID:      execution.UserID,
+		JobName:     execution.JobName,
+		UserName:    execution.UserName,
+		FilePresent: execution.FilePresent,
+	}
+
+	return operation.NewGetScanDataExportExecutionOK().WithPayload(&sdeExec)
+}
+
+func (se *scanDataExportAPI) DownloadScanData(ctx context.Context, params operation.DownloadScanDataParams) middleware.Responder {
+	err := se.RequireAuthenticated(ctx)
```

Comment:
does that mean anyone can download CVE data of any exeuction?

Reply:
Good catch. I missed this in my earlier implementation. Perhaps because the userID is set to -1, the above scenario pointed out is feasible. 
One fix would be to filter the executions based on the name of the user triggering the request and return only those execution records or download data that belong to the user
The other alternative is what requires your input - send the user id as the vendor id in which case, we return the executions filtered by the user id.

Reply:
Done. I have implemented a check against the name of the user initiating the download versus the name of the user who owns the task execution and return a 401 if they do not match